docs/guide/interactions/modals.rst

```diff
@@ -0,0 +1,143 @@
+the constructor. The title is required and is displayed at the top of the modal.
+
+
+Fields
+~~~~~~~
+
+In our example above we have three class-attributes: ``name``, ``feedback``, and ``additional_information``, all of which are instances of the :class:`~discord.ui.TextInput` class.
+In modal classes, class-attributes (which are instances of components) are used to represent the fields of the modal.
+
+Discord requires modals contain between 1 and 5 fields.
+
+.. note:: 
+
+    Modal fields are displayed in the order in which they were defined. Our example from above defines ``name`` before it defines ``feedback``, 
+    and that order is preserved when the ``FeedbackForm`` modal is displayed on Discord.
+
+Fields can also be added to or removed from a modal instance using the :meth:`~discord.ui.Modal.add_item` and :meth:`~discord.ui.Modal.remove_item` methods.
+
+.. warning:: 
+
+    Ensure that class-attribute defined fields do not conflict with attributes or methods defined in the :class:`parent class <discord.ui.Modal>`, 
+    as this will cause unexpected behaviour.
+
+
+You can customize individual modal instances with normal attribute access and assignment.
+For example, to change the label of the ``name`` field to ``'Your Name'`` you can do the following:
+
+.. code-block:: python
+
+    form = FeedbackForm()
+    form.name.label = 'Your Name'
+
+
+Handling User Responses
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Once the user has clicked `Submit` the :meth:`Modal.on_submit <discord.ui.Modal.on_submit>` callback is called. 
+The callback is passed a new :class:`~discord.Interaction` instance, which requires a response.
+
+In the example above we respond by sending a message to the user confirming their submission was recorded.
+
+
+Handling an error
+~~~~~~~~~~~~~~~~~~
+
+The :meth:`Modal.on_error <discord.ui.Modal.on_error>` method is called when an exception is raised
+within :meth:`Modal.on_submit <discord.ui.Modal.on_submit>`, and can be used to handle an error or respond to the user.
+
+This method is passed the exception raised, and the :class:`~discord.Interaction` instance, allowing you to handle the error and, 
+if necessary, send a response to the user.
+
+
+.. code-block:: python
+    :emphasize-lines: 13-15
+
+    class FeedbackForm(discord.ui.Modal, title='Feedback'):
+        name = discord.ui.TextInput(label='Name')
+        feedback = discord.ui.TextInput(label='Feedback', style=discord.TextStyle.long)
+        additional_information = discord.ui.TextInput(
+            label='Additional Information', style=discord.TextStyle.long, required=False
+        )
+
+        async def on_submit(self, interaction: discord.Interaction) -> None:
+            ... # do something with the data
+
+            await interaction.response.send_message(f'Thank you {self.name.value}, your submission was recorded.')
+
+        async def on_error(self, error: Exception, interaction: discord.Interaction) -> None:
```

Comment:
The arguments for `on_error` are incorrect. The proper ones should be noted below:

`(self, interaction: discord.Interaction, error: Exception)`